### ISSUE TITLE
Rule proposal: No unnecessary `Promise.resolve/reject()` in async functions

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#1609]
### Description

The rule should prevent using moot `Promise.reject` and `Promise.resovle` calls in async functions.

Rule name suggestion: `no-useless-promise-resolve-reject`

### Fail

```js
const main = async foo => {
	if (foo > 4) {
		return Promise.reject(new Error('🤪'));
	}

	return Promise.resolve(result);
};
```

### Pass

```js
const main = async foo => {
	if (foo > 4) {
		throw new Error('🤪');
	}

	return result;
};
```

**Comment by fisker** [comment]
Should it apply to async generator function? I'm not sure how it works. But I assume they result the same?

**Comment by sindresorhus** [comment]
Yes

**Comment by j-f1** [comment]
Another place this should apply: `Promise.then`/`.catch` callbacks.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
I've seen people also try to wrap a thenable with Promise(both sync and async functions)

```js
function foo() {
  return new Promise((resolve, reject) => {
    bar.then(resolve, reject);
  })
}
```

Especially, old projects uses `jQuery.ajax()`, they thought it has to be a native Promise, but in most cases, the thenable works the same as Promises.

Though this didn't use `Promise.{resolve,reject}`, this case is also "unnecessary promise".

**Comment by fisker** [comment]
Seems we missed this https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1609#issuecomment-970571215

**Comment by fisker** [comment]
@cherryblossom000 ^
